vortex-api/src/main/java/mil/army/usace/hec/vortex/temporal/PeriodRecordIndexQuery.java

```diff
@@ -0,0 +1,49 @@
+package mil.army.usace.hec.vortex.temporal;
+
+import java.time.ZonedDateTime;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+final class PeriodRecordIndexQuery implements RecordIndexQuery {
+    private final Map<VortexTimeRecord, Integer> originalIndexMap;
+    private final StaticIntervalTree<VortexTimeRecord> intervalTree;
+
+    PeriodRecordIndexQuery(List<VortexTimeRecord> recordList) {
```

Comment:
I usually pass args as an array so that type can be enforced.

Reply:
https://github.com/HydrologicEngineeringCenter/Vortex/pull/116#discussion_r1791894175